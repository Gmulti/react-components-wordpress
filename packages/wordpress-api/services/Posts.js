import Endpoint from './shared/Endpoint'

class Posts extends Endpoint {
    get endpoint() {
        return '/posts'
    }
}

export default Posts
