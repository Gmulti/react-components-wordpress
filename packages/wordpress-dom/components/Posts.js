import React, { Component } from 'react'
import connectPosts from '../../wordpress-api/connectors/connectPosts'

@connectPosts()
class Posts extends Component {
    render() {
        const { posts = [] } = this.props
        return (
            <ul>
                {posts.map((post, key) => (
                    <li key={key}>{post.title.rendered}</li>
                ))}
            </ul>
        )
    }
}

export default Posts
