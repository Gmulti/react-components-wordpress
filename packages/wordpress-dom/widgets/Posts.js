import React, { useEffect } from 'react'
import { connectPosts } from '../../wordpress-api/connectors/connectPosts'
import { withPostsProvider, withPosts } from '../../wordpress-context/withPosts'
import PostsComponent from '../ui/Posts'
import { isNull } from 'lodash'

const Posts = props => {
    const {
        postsContext: { dispatch, posts },
        getServicePostsApi
    } = props

    useEffect(() => {
        const fetchData = async () => {
            const posts = await getServicePostsApi().all()
            if (!isNull(posts)) {
                dispatch({ type: 'addMany', payload: posts })
            }
        }

        fetchData()
    }, [])

    return <PostsComponent items={posts} {...props} />
}

export default connectPosts(withPostsProvider()(withPosts()(Posts)))
