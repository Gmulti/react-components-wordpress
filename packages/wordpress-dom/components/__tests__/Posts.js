import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import Posts from '../Posts'

Enzyme.configure({ adapter: new Adapter() })

describe('Posts', () => {
    it('expect to render', () => {
        const wrapper = shallow(<Posts />)

        expect(wrapper).toMatchSnapshot()
    })

    it('expect to render with props', () => {
        const instance = renderer.create(
            <Posts
                posts={[
                    {
                        title: {
                            rendered: 'Hello World'
                        }
                    },
                    {
                        title: {
                            rendered: 'Foo Bar'
                        }
                    }
                ]}
            />
        )

        expect(instance.toJSON()).toMatchSnapshot()
    })
})
