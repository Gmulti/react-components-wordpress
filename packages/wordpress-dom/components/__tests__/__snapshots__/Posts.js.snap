// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts expect to render 1`] = `<ul />`;

exports[`Posts expect to render with props 1`] = `
<ul>
  <li>
    Hello World
  </li>
  <li>
    Foo Bar
  </li>
</ul>
`;
